beekeeper-path-scheduler-apiary/src/main/java/com/expediagroup/beekeeper/scheduler/apiary/messaging/MessageReaderAdapter.java

```diff
@@ -16,38 +16,58 @@
 package com.expediagroup.beekeeper.scheduler.apiary.messaging;
 
 import java.io.IOException;
+import java.util.List;
 import java.util.Optional;
+import java.util.stream.Collectors;
 
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 import com.expedia.apiary.extensions.receiver.common.messaging.MessageEvent;
 import com.expedia.apiary.extensions.receiver.common.messaging.MessageReader;
 
-import com.expediagroup.beekeeper.scheduler.apiary.model.PathEvent;
+import com.expediagroup.beekeeper.core.model.HousekeepingPath;
+import com.expediagroup.beekeeper.scheduler.apiary.handler.MessageEventHandler;
+import com.expediagroup.beekeeper.scheduler.apiary.model.BeekeeperEvent;
 
-public class MessageReaderAdapter implements PathEventReader {
+public class MessageReaderAdapter implements BeekeeperEventReader {
 
   private static final Logger log = LoggerFactory.getLogger(MessageReaderAdapter.class);
 
   private final MessageReader delegate;
-  private final MessageEventToPathEventMapper mapper;
+  private final List<MessageEventHandler> handlers;
 
-  public MessageReaderAdapter(MessageReader delegate, MessageEventToPathEventMapper mapper) {
+  public MessageReaderAdapter(MessageReader delegate, List<MessageEventHandler> handlers) {
     this.delegate = delegate;
-    this.mapper = mapper;
+    this.handlers = handlers;
   }
 
   @Override
-  public Optional<PathEvent> read() {
+  public Optional<BeekeeperEvent> read() {
     Optional<MessageEvent> messageEvent = delegate.read();
-    return messageEvent.flatMap(mapper::map);
+
+    if (!messageEvent.isPresent()) {
+      return Optional.empty();
+    }
+
+    MessageEvent message = messageEvent.get();
+
+    List<HousekeepingPath> housekeepingPaths = handlers.parallelStream()
+        .map(eventHandler -> eventHandler.handleMessage(message))
+        .flatMap(x -> x.stream())
+        .collect(Collectors.toList());
+
+    if (housekeepingPaths.size() <= 0) {
```

Comment:
If there are no housekeeping paths, we can return an empty optional, but we still need to remove the event from the queue